### PR TITLE
QR generator

### PR DESCRIPTION
**Pull request by Gerussi**
### What?

A new button as been added to the profile view, enabling to generate a QR Code that would allow other people to easily follow this profile.

### CODE REVIEW

**Review by atappy — COMMENTED** [review]
Great Job ! Is it possible to add some tests and add the possibility to share the QR code outside of the app ?

**Comment by Gerussi** [comment]
> Great Job ! Is it possible to add some tests and add the possibility to share the QR code outside of the app ?

This will be done in other PRs. Firstly the reading and proper behaviour will be implemented in the next PR. Once the basic feature work properly, the tests will be added. Finally, any extra features (colors changes, sharing without needed to screen,...) will be added if time permits it

**Review by atappy — APPROVED** [review]

File: app/src/main/java/com/github/palFinderTeam/palfinder/ProfileActivity.kt [hunk 1]
```diff
@@ -126,6 +136,26 @@ class ProfileActivity : AppCompatActivity() {
...
     }
 
+    /**
+     * Clicking on QR Code icon will show QR code
+     */
+    fun showQR(view: View?) {
+        //Initiate the barcode encoder
+        val barcodeEncoder = BarcodeEncoder()
+        //Encode text in editText into QRCode image into the specified size using barcodeEncoder
+        val bitmap = barcodeEncoder.encodeBitmap(intent.getStringExtra(USER_ID), BarcodeFormat.QR_CODE, 512, 512)
```
Comment (TruddyTheDuddi) [inline app/src/main/java/com/github/palFinderTeam/palfinder/ProfileActivity.kt:146]: Same comments as on #299, variable could be reused!

**Review by TruddyTheDuddi — APPROVED** [review]
Nice!

**Pull request merged**